web/modules/custom/joinup_invite/src/Form/InviteToGroupForm.php

```diff
@@ -98,33 +131,168 @@ public function buildForm(array $form, FormStateInterface $form_state, RdfInterf
+
   /**
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
     $users = $this->getUserList($form_state);
-    $role_id = implode('-', [
-      $this->rdfEntity->getEntityTypeId(),
-      $this->rdfEntity->bundle(),
-      $form_state->getValue('role'),
-    ]);
+    $role_option = $form_state->getValue('role');
+    $role_id = $this->getRoleId($role_option);
     $role = $this->entityTypeManager->getStorage('og_role')->load($role_id);
+    $current_user = $this->entityTypeManager->getStorage('user')->load($this->currentUser->id());
 
     foreach ($users as $user) {
-      $membership = $this->ogMembershipManager->getMembership($this->rdfEntity, $user);
-      if (empty($membership)) {
-        $membership = $this->ogMembershipManager->createMembership($this->rdfEntity, $user);
+      if ($this->rdfEntity->bundle() === 'collection') {
+        $membership = $this->ogMembershipManager->createMembership($this->rdfEntity, $user)
+          ->addRole($role)
+          ->setState(OgMembershipInterface::STATE_PENDING);
+        // Skip notifications related to memberships.
+        $membership->skip_notification = TRUE;
+        $membership->save();
+
+        /** @var \Drupal\joinup_invite\Entity\InvitationInterface $invitation */
+        $invitation = $this->entityTypeManager->getStorage('invitation')
+          ->create(['bundle' => 'group'])
+          ->setEntity($this->rdfEntity)
+          ->setRecipient($user)
+          ->setOwner($current_user)
+          ->setStatus(InvitationInterface::STATUS_PENDING);
+        $invitation->save();
+
+        $this->sendMessage($invitation, $role_option);
+        drupal_set_message($this->t('An invitation has been sent to the selected users. Their membership is pending.'));
+      }
+      // @todo: Remove this when the invitations for solutions are implemented.
+      elseif ($this->rdfEntity->bundle() === 'solution') {
+        $membership = $this->ogMembershipManager->getMembership($this->rdfEntity, $user);
+        if (empty($membership)) {
+          $membership = $this->ogMembershipManager->createMembership($this->rdfEntity, $user);
+        }
+        $membership->addRole($role);
+        $membership->save();
+
+        drupal_set_message($this->t('Successfully added the role %role to the selected users.', [
+          '%role' => $role->label(),
+        ]));
       }
-      $membership->addRole($role);
-      $membership->save();
     }
 
-    drupal_set_message($this->t('Successfully added the role %role to the selected users.', [
-      '%role' => $role->label(),
-    ]));
     $form_state->setRedirect('entity.rdf_entity.member_overview', [
       'rdf_entity' => $this->rdfEntity->id(),
     ]);
   }
 
+  /**
+   * Returns the og role id of the current group.
+   *
+   * @param string $role_option
+   *   The role machine name.
+   *
+   * @return string
+   *   The role id.
+   */
+  protected function getRoleId($role_option): string {
```

Comment:
It only makes the code a bit more readable. Same as for the next method (send message). But since the variables would require to break the array into multiple lines, and the code is already complex (since we only implement the collection invitation and the solution remains as is), I thought it improves visibility.

Reply:
OK